### ISSUE TITLE
Disallow `!??` in TypeScript

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#918]
`!` in TypeScript (non-null assertion operator) cancels the null type and `??` (nullish coalescing operator) actually handles it. The first one is redundant

## Fail

```js
return document.querySelector('a')! ?? defaultElement
```

## Pass

```js
return document.querySelector('a') ?? defaultElement
```

**Comment by fregante** [comment]
This however should probably be part of the official TypeScript plugin since I think it’s likely to be accepted there.

**Comment by sindresorhus** [comment]
> This however should probably be part of the official TypeScript plugin since I think it’s likely to be accepted there. 

Yup

**Comment by fregante** [comment]
Done and accepted
